hddfancontrol: 1.5.1->2.0.1

jadewilk opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

Updated to 2.0.1. [Migration guide](https://github.com/desbma/hddfancontrol?tab=readme-ov-file#migrating-from-v1x). v2 included a rust rewrite.
nixos test does not seem feasible anymore as it now tries to get the model name of the disks, which is not possible with the vm disk (maybe can spoof it? idk)
![image](https://github.com/user-attachments/assets/7e7c8860-23f6-44d1-b9ea-188919bd0b64)


<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- ❌  Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

@benley 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

JohnRTitor commented:
> maybe can spoof it? idk)

if you can sure, but if the test can't work on a test vm at all, I would root for removing it.

SuperSandro2000 commented on nixos/doc/manual/release-notes/rl-2505.section.md:
```diff
@@ -528,6 +528,8 @@
 
 - `services.avahi.ipv6` now defaults to true.
 
+- hddfancontrol has been updated to 2.0.1. See the [migration guide](https://github.com/desbma/hddfancontrol/tree/master?tab=readme-ov-file#migrating-from-v1x), as there are breaking changes.
```
```suggestion
- hddfancontrol has been updated to major release 2. See the [migration guide](https://github.com/desbma/hddfancontrol/tree/master?tab=readme-ov-file#migrating-from-v1x), as there are breaking changes.
```
We don't want to care if 2.0.2 is released by Mai.

SuperSandro2000 commented on nixos/modules/services/hardware/hddfancontrol.nix:
```diff
@@ -7,16 +7,22 @@
+      "services"
+      "hddfancontrol"
+      "smartctl"
+    ] "This argument has been removed upstream, and smartctl is automatically used when necessary.")
```
```suggestion
    ] "smartctl is now automatically used when necessary which makes the option redundant.")
```

SuperSandro2000 commented on pkgs/by-name/hd/hddfancontrol/package.nix:
```diff
@@ -1,45 +1,55 @@
 
   src = fetchFromGitHub {
     owner = "desbma";
     repo = pname;
```
```suggestion
    repo = "hddfancontrol";
```

JohnRTitor reviewed (commented):
Please remove the test in a separate commit, with a concise message explaining why this is necessary.

JohnRTitor reviewed (commented):
Split out the module change into a separate commit as well, should start with `nixos/hddfancontrol`.

The release notes change can stick together with the package update commit.

JohnRTitor commented on pkgs/by-name/hd/hddfancontrol/package.nix:
```diff
@@ -1,45 +1,55 @@
 }:
-
-python3Packages.buildPythonPackage rec {
+rustPlatform.buildRustPackage rec {
```
Please use the finalAttrs pattern, Rust builders support it now.

JohnRTitor commented on pkgs/by-name/hd/hddfancontrol/package.nix:
```diff
@@ -1,45 +1,55 @@
+      }
+  '';
 
   meta = with lib; {
```
Remove the `with lib` too while you are at it.

JohnRTitor commented on pkgs/by-name/hd/hddfancontrol/package.nix:
```diff
@@ -1,45 +1,55 @@
 
   meta = with lib; {
     description = "Dynamically control fan speed according to hard drive temperature on Linux";
     homepage = "https://github.com/desbma/hddfancontrol";
```
Set changelog to https://github.com/desbma/hddfancontrol/releases

JohnRTitor commented on pkgs/by-name/hd/hddfancontrol/package.nix:
```diff
@@ -1,45 +1,65 @@
-    rev = version;
-    sha256 = "0b2grf98qnikayn18xll01dkm5pjpcjxdffgx1nyw9s0gqig8dg0";
+    repo = "hddfancontrol";
+    rev = finalAttrs.version;
```
```suggestion
    tag = finalAttrs.version;
```

jadewilk committed:
hddfancontrol: 1.5.1->2.0.1

JohnRTitor commented on nixos/modules/services/hardware/hddfancontrol.nix:
Consider adding yourself to meta.maintainers of this module as well.

JohnRTitor reviewed (approved):
Rest LGTM

jadewilk committed:
nixos/hddfancontrol: support v2 argument format

jadewilk committed:
nixosTests.hddfancontrol: remove

Version 2 of hddfancontrol now fetches disk info - namely model - from the passed disks immediately, which /dev/vda does not have.
This is a fatal error that exits the daemon, so the test is no longer possible.
